src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -102,18 +102,37 @@ internal static partial class ICodeDefinitionFactoryExtensions
                 // Special case.  If we're generating a struct constructor, then we'll need
                 // to initialize all fields in the struct, not just the ones we're creating.  To
                 // do that, we call the default constructor.
-                var realFields = containingTypeOpt.GetMembers()
-                                     .OfType<IFieldSymbol>()
-                                     .Where(f => !f.IsStatic);
-                var initializedFields = parameterToExistingFieldMap.Values
-                                            .OfType<IFieldSymbol>()
-                                            .Where(f => !f.IsImplicitlyDeclared && !f.IsStatic);
-                if (initializedFields.Count() < realFields.Count())
+
+                var numFieldsRequiringInitialization = 0;
+
+                var autoProperties = new List<IPropertySymbol>();
+                foreach (var containingTypeField in containingTypeOpt.GetMembers().OfType<IFieldSymbol>())
+                {
+                    if (containingTypeField.IsStatic) continue;
```

Comment:
curlies always required

Reply:
:+1:
Seems like something that the codebase should enforce via analyzer to save time. @sharwell?

Reply:
yes & they should definitely mention it here: https://github.com/dotnet/roslyn/wiki/Contributing-Code#coding-conventions

Reply:
@Neme12 https://github.com/dotnet/roslyn/wiki/Contributing-Code#coding-conventions first item says: Use the coding style outlined in the [.NET Foundation Coding Guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md)

First rule there says: 

> A single line statement block can go without braces but the block must be properly indented on its own line and it must not be nested in other statement blocks that use braces

First code sample shows:

```cs
            if (items == null)
                throw new ArgumentNullException(nameof(items));
```

Reply:
@jnm2  I know, I was hit by this too.

Reply:
> Seems like something that the codebase should enforce via analyzer to save time.

I'll get this fixed

Reply:
Curlies always requied.  and body must be on the next line.  i.e. not  ```if (foo) { continue; }```.